### ISSUE TITLE
vertical positioning of rests 

### ISSUE THREAD

**Issue by rettinghaus** [rism-ch/verovio#344]
According to Gould rests are put to the middle of the staff. When using a G clef the middle line represents b4. So rests pointed to that very position should be expected getting drawn in their *normal* place. 
Then d5 means one line above, g4 one line below, etc.  
This is what LilyPond and MuseScore do (rests on d5 - b4 - g4 - e4 respectively): 

LilyPond:
![rests lilypond](https://cloud.githubusercontent.com/assets/7693447/19967861/48ebd350-a1d2-11e6-9f3c-ae4ac600e55e.png)

MuseScore:
![rests musescore](https://cloud.githubusercontent.com/assets/7693447/19967862/48ec3aca-a1d2-11e6-8c2d-77f45c1623d3.png)

Verovio puts the half rests in the expected positions, but all shorter rests get drawn one line (or 2 vu) too high. 

Verovio:
![rests verovio](https://cloud.githubusercontent.com/assets/7693447/19968141/9793add8-a1d3-11e6-9979-0beda871a157.png)

## Code
LilyPond:
```
\relative c'' { d1\rest b\rest g\rest e\rest d'2\rest b\rest g\rest e\rest d'4\rest b\rest g\rest e\rest d'8\rest b\rest g\rest e\rest d'16\rest b\rest g\rest e\rest d'32\rest b\rest g\rest e\rest d'64\rest b\rest g\rest e\rest d'128\rest b\rest g\rest e\rest }
```

MusicXML:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 3.0 Partwise//EN" "http://www.musicxml.org/dtds/partwise.dtd">
<score-partwise>
  <part-list>
    <score-part id="P1">
      <part-name></part-name>
      <part-abbreviation></part-abbreviation>
      <score-instrument id="P1-I1">
        <instrument-name></instrument-name>
      </score-instrument>
      <midi-device id="P1-I1" port="1">
      </midi-device>
    </score-part>
  </part-list>
  <part id="P1">
    <measure number="1" width="310.45">
      <print>
        <system-layout>
          <system-margins>
            <left-margin>0.00</left-margin>
            <right-margin>-0.00</right-margin>
          </system-margins>
          <top-system-distance>70.00</top-system-distance>
        </system-layout>
      </print>
      <attributes>
        <divisions>32</divisions>
        <key>
          <fifths>0</fifths>
        </key>
        <time>
          <beats>4</beats>
          <beat-type>4</beat-type>
        </time>
        <clef>
          <sign>G</sign>
          <line>2</line>
        </clef>
      </attributes>
      <note>
        <rest>
          <display-step>D</display-step>
          <display-octave>5</display-octave>
        </rest>
        <duration>128</duration>
        <voice>1</voice>
      </note>
    </measure>
    <measure number="2" width="255.68">
      <note>
        <rest>
          <display-step>B</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>128</duration>
        <voice>1</voice>
      </note>
    </measure>
    <measure number="3" width="255.68">
      <note>
        <rest>
          <display-step>G</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>128</duration>
        <voice>1</voice>
      </note>
    </measure>
    <measure number="4" width="255.68">
      <note>
        <rest>
          <display-step>E</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>128</duration>
        <voice>1</voice>
      </note>
    </measure>
    <measure number="5" width="255.68">
      <note>
        <rest>
          <display-step>D</display-step>
          <display-octave>5</display-octave>
        </rest>
        <duration>64</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
      <note>
        <rest>
          <display-step>B</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>64</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
    </measure>
    <measure number="5" width="255.68">
      <note>
        <rest>
          <display-step>G</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>64</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
      <note>
        <rest>
          <display-step>E</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>64</duration>
        <voice>1</voice>
        <type>half</type>
      </note>
    </measure>
    <measure number="6" width="252.16">
      <note>
        <rest>
          <display-step>D</display-step>
          <display-octave>5</display-octave>
        </rest>
        <duration>32</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <note>
        <rest>
          <display-step>B</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>32</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <note>
        <rest>
          <display-step>G</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>32</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
      <note>
        <rest>
          <display-step>E</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>32</duration>
        <voice>1</voice>
        <type>quarter</type>
      </note>
    </measure>
    <measure number="7" width="566.23">
      <note>
        <rest>
          <display-step>D</display-step>
          <display-octave>5</display-octave>
        </rest>
        <duration>16</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest>
          <display-step>B</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>16</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest>
          <display-step>G</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>16</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest>
          <display-step>E</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>16</duration>
        <voice>1</voice>
        <type>eighth</type>
      </note>
      <note>
        <rest>
          <display-step>D</display-step>
          <display-octave>5</display-octave>
        </rest>
        <duration>8</duration>
        <voice>1</voice>
        <type>16th</type>
      </note>
      <note>
        <rest>
          <display-step>B</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>8</duration>
        <voice>1</voice>
        <type>16th</type>
      </note>
      <note>
        <rest>
          <display-step>G</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>8</duration>
        <voice>1</voice>
        <type>16th</type>
      </note>
      <note>
        <rest>
          <display-step>E</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>8</duration>
        <voice>1</voice>
        <type>16th</type>
      </note>
      <note>
        <rest>
          <display-step>D</display-step>
          <display-octave>5</display-octave>
        </rest>
        <duration>4</duration>
        <voice>1</voice>
        <type>32nd</type>
      </note>
      <note>
        <rest>
          <display-step>B</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>4</duration>
        <voice>1</voice>
        <type>32nd</type>
      </note>
      <note>
        <rest>
          <display-step>G</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>4</duration>
        <voice>1</voice>
        <type>32nd</type>
      </note>
      <note>
        <rest>
          <display-step>E</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>4</duration>
        <voice>1</voice>
        <type>32nd</type>
      </note>
      <note>
        <rest>
          <display-step>D</display-step>
          <display-octave>5</display-octave>
        </rest>
        <duration>2</duration>
        <voice>1</voice>
        <type>64th</type>
      </note>
      <note>
        <rest>
          <display-step>B</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>2</duration>
        <voice>1</voice>
        <type>64th</type>
      </note>
      <note>
        <rest>
          <display-step>G</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>2</duration>
        <voice>1</voice>
        <type>64th</type>
      </note>
      <note>
        <rest>
          <display-step>E</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>2</duration>
        <voice>1</voice>
        <type>64th</type>
      </note>
      <note>
        <rest>
          <display-step>D</display-step>
          <display-octave>5</display-octave>
        </rest>
        <duration>1</duration>
        <voice>1</voice>
        <type>128th</type>
      </note>
      <note>
        <rest>
          <display-step>B</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>1</duration>
        <voice>1</voice>
        <type>128th</type>
      </note>
      <note>
        <rest>
          <display-step>G</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>1</duration>
        <voice>1</voice>
        <type>128th</type>
      </note>
      <note>
        <rest>
          <display-step>E</display-step>
          <display-octave>4</display-octave>
        </rest>
        <duration>1</duration>
        <voice>1</voice>
        <type>128th</type>
      </note>
    </measure>
  </part>
</score-partwise>
```

MEI:
```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<?xml-model href="https://raw.githubusercontent.com/music-encoding/music-encoding/master/schemata/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://raw.githubusercontent.com/music-encoding/music-encoding/master/schemata/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei"
     xmlns:xlink="http://www.w3.org/1999/xlink"
     meiversion="3.0.0">
   <meiHead>
      <fileDesc>
         <titleStmt>
            <title/>
         </titleStmt>
         <pubStmt/>
         <sourceDesc>
            <source>
               <titleStmt>
                  <title/>
               </titleStmt>
               <pubStmt/>
            </source>
         </sourceDesc>
      </fileDesc>
      <encodingDesc>
      </encodingDesc>
   </meiHead>
   <music>
      <body>
         <mdiv>
            <score>
               <scoreDef meter.count="4" meter.unit="4" key.sig="0" music.size="2in">
                  <staffGrp>
                     <staffDef n="1" xml:id="P1" lines="5" clef.line="2" clef.shape="G"/>
                  </staffGrp>
               </scoreDef>
               <section>
                  <measure n="1" xml:id="d1e52">
                     <staff n="1">
                        <layer n="1">
                           <rest xml:id="d1e79" dur="1" ploc="d" oloc="5"/>
                        </layer>
                     </staff>
                  </measure>
                  <measure n="2" xml:id="d1e89">
                     <staff n="1">
                        <layer n="1">
                           <rest xml:id="d1e91" dur="1" ploc="b" oloc="4"/>
                        </layer>
                     </staff>
                  </measure>
                  <measure n="3" xml:id="d1e101">
                     <staff n="1">
                        <layer n="1">
                           <rest xml:id="d1e103" dur="1" ploc="g" oloc="4"/>
                        </layer>
                     </staff>
                  </measure>
                  <measure n="4" xml:id="d1e113">
                     <staff n="1">
                        <layer n="1">
                           <rest xml:id="d1e115" dur="1" ploc="e" oloc="4"/>
                        </layer>
                     </staff>
                  </measure>
                  <measure n="5" xml:id="d1e125">
                     <staff n="1">
                        <layer n="1">
                           <rest xml:id="d1e127" dur="2" ploc="d" oloc="5"/>
                           <rest xml:id="d1e139" dur="2" ploc="b" oloc="4"/>
                        </layer>
                     </staff>
                  </measure>
                  <measure n="5" xml:id="d1e176">
                     <staff n="1">
                        <layer n="1">
                           <rest xml:id="d1e203" dur="2" ploc="g" oloc="4"/>
                           <rest xml:id="d1e215" dur="2" ploc="e" oloc="4"/>
                        </layer>
                     </staff>
                  </measure>
                  <measure n="6" xml:id="d1e227">
                     <staff n="1">
                        <layer n="1">
                           <rest xml:id="d1e229" dur="4" ploc="d" oloc="5"/>
                           <rest xml:id="d1e241" dur="4" ploc="b" oloc="4"/>
                           <rest xml:id="d1e253" dur="4" ploc="g" oloc="4"/>
                           <rest xml:id="d1e265" dur="4" ploc="e" oloc="4"/>
                        </layer>
                     </staff>
                  </measure>
                  <measure n="7" xml:id="d1e277">
                     <staff n="1">
                        <layer n="1">
                           <rest xml:id="d1e279" dur="8" ploc="d" oloc="5"/>
                           <rest xml:id="d1e291" dur="8" ploc="b" oloc="4"/>
                           <rest xml:id="d1e303" dur="8" ploc="g" oloc="4"/>
                           <rest xml:id="d1e315" dur="8" ploc="e" oloc="4"/>
                           <rest xml:id="d1e327" dur="16" ploc="d" oloc="5"/>
                           <rest xml:id="d1e339" dur="16" ploc="b" oloc="4"/>
                           <rest xml:id="d1e351" dur="16" ploc="g" oloc="4"/>
                           <rest xml:id="d1e363" dur="16" ploc="e" oloc="4"/>
                           <rest xml:id="d1e375" dur="32" ploc="d" oloc="5"/>
                           <rest xml:id="d1e387" dur="32" ploc="b" oloc="4"/>
                           <rest xml:id="d1e399" dur="32" ploc="g" oloc="4"/>
                           <rest xml:id="d1e412" dur="32" ploc="e" oloc="4"/>
                           <rest xml:id="d1e424" dur="64" ploc="d" oloc="5"/>
                           <rest xml:id="d1e436" dur="64" ploc="b" oloc="4"/>
                           <rest xml:id="d1e448" dur="64" ploc="g" oloc="4"/>
                           <rest xml:id="d1e460" dur="64" ploc="e" oloc="4"/>
                           <rest xml:id="d1e472" dur="128" ploc="d" oloc="5"/>
                           <rest xml:id="d1e484" dur="128" ploc="b" oloc="4"/>
                           <rest xml:id="d1e496" dur="128" ploc="g" oloc="4"/>
                           <rest xml:id="d1e508" dur="128" ploc="e" oloc="4"/>
                        </layer>
                     </staff>
                  </measure>
               </section>
            </score>
         </mdiv>
      </body>
   </music>
</mei>
```

**Comment by rettinghaus** [comment]
*Normal* positioning of rests with shorter durations doesn't work correctly either. 
![rests norm](https://cloud.githubusercontent.com/assets/7693447/20003055/71899d62-a284-11e6-8b6d-b7a3782c84f2.png)

Flagged rests shouldn't collide with staff lines and dots should be aligned to the top of flagged rests. 
It should look something like this: 
![restsnorm preview](https://cloud.githubusercontent.com/assets/7693447/20003214/9b867788-a285-11e6-8fcb-129338f54d63.png)

```xml
<mei xmlns="http://www.music-encoding.org/ns/mei" xmlns:xlink="http://www.w3.org/1999/xlink" meiversion="3.0.0">
  <meiHead>
  </meiHead>
  <music>
    <body>
      <mdiv>
        <score>
          <scoreDef key.sig="0" key.mode="major">
            <staffGrp n="1" >
              <staffDef n="1" lines="5" clef.line="2" clef.shape="G"/>
            </staffGrp>
          </scoreDef>
          <section>
            <measure n="1" right="dbl">
              <staff n="1">
                <layer n="1">
                  <rest dur="2"/>
                  <rest dur="4"/>
                  <rest dur="8"/>
                  <rest dur="16"/>
                  <rest dur="32"/>
                  <rest dur="64"/>
                  <rest dur="128"/>
                  <rest dur="256"/>
                </layer>
              </staff>
            </measure>
            <measure n="1" right="dbl">
              <staff n="1">
                <layer n="1">
                  <rest dur="2" dots="1"/>
                  <rest dur="4" dots="1"/>
                  <rest dur="8" dots="1"/>
                  <rest dur="16" dots="1"/>
                  <rest dur="32" dots="1"/>
                  <rest dur="64" dots="1"/>
                  <rest dur="128" dots="1"/>
                  <rest dur="256" dots="1"/>
                </layer>
              </staff>
            </measure>
          </section>
        </score>
      </mdiv>
    </body>
  </music>
</mei>
```

**Comment by rettinghaus** [comment]
The last example is a Leipzig font only problem. The shorter rests should be reworked.

**Comment by lpugin** [comment]
Fixed with 8379808064c5006fb8967ae87e1116efef91e222

![image](https://cloud.githubusercontent.com/assets/689412/25502626/a99de3ce-2b97-11e7-991c-13c014fa2760.png)
